controllers/volumereplicationgroup_controller.go

```diff
@@ -753,55 +779,52 @@ func (v *VRGInstance) processAsPrimary() (ctrl.Result, error) {
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	if err := v.restorePVs(); err != nil {
+	result := ctrl.Result{}
+	if err := v.restorePVs(&result); err != nil {
 		v.log.Info("Restoring PVs failed", "Error", err.Error())
 
 		msg := fmt.Sprintf("Failed to restore PVs (%v)", err.Error())
 		setVRGClusterDataErrorCondition(&v.instance.Status.Conditions, v.instance.Generation, msg)
 
 		if err = v.updateVRGStatus(false); err != nil {
 			v.log.Error(err, "VRG Status update failed")
+
+			result.Requeue = true
 		}
 
-		// Since updating status failed, reconcile
-		return ctrl.Result{Requeue: true}, nil
+		return result, nil
 	}
 
-	requeue := v.handleVRGMode(ramendrv1alpha1.Primary)
+	result = v.reconcileAsPrimary()
 
 	// If requeue is false, then VRG was successfully processed as primary.
 	// Hence the event to be generated is Success of type normal.
 	// Expectation is that, if something failed and requeue is true, then
 	// appropriate event might have been captured at the time of failure.
-	if !requeue {
+	if !result.Requeue {
 		rmnutil.ReportIfNotPresent(v.reconciler.eventRecorder, v.instance, corev1.EventTypeNormal,
 			rmnutil.EventReasonPrimarySuccess, "Primary Success")
 	}
 
 	if err := v.updateVRGStatus(true); err != nil {
-		requeue = true
-	}
-
-	if requeue {
-		v.log.Info("Requeuing resource as Primary")
-
-		return ctrl.Result{Requeue: true}, nil
+		result.Requeue = true
 	}
 
-	v.log.Info("Successfully processed vrg as primary")
+	delayResetIfRequeueTrue(&result, v.log)
 
-	return ctrl.Result{}, nil
+	return result, nil
 }
 
-func (v *VRGInstance) reconcileAsPrimary() bool {
-	requeueForVolSync := false
+func (v *VRGInstance) reconcileAsPrimary() ctrl.Result {
+	result := ctrl.Result{}
 	if len(v.volSyncPVCs) != 0 {
-		requeueForVolSync = v.reconcileVolSyncAsPrimary()
+		result.Requeue = v.reconcileVolSyncAsPrimary()
 	}
 
-	requeueForVolRep := v.reconcileVolRepsAsPrimary()
+	v.reconcileVolRepsAsPrimary(&result.Requeue)
+	v.kubeObjectsProtect(&result)
```

Comment:
One of the first things the `kubeObjectsProtect` function, which is called on line 825, does is check if the `kubeObjectProtection` is enabled.  However, the VRG object upload is attempted even if the `kubeObjectProtection` is disabled and will result in the `ClusterDataProtected` being not true if there are no s3 stores.  I presume this is what is preventing the relocation.  I think this VRG object upload can also set its `ClusterDataProtected` subcondition to true or nil if the special "NoS3" profile is listed and it would have been false otherwise.  I intend to prioritize this change.

Regarding a longer-term solution: a new condition could be introduced for the VRG object protection such that it is not included in the `ClusterDataProtected` condition, though I think it should be included, so perhaps DRPC could not require `ClusterDataProtected` be true to failback if only volsync volumes are selected.

Reply:
This has been addressed by skipping the `NoS3StoreAvailable` profile for VRG and Velero backup uploads.  VRG and Velero now submit a `nil` `ClusterDataProtected` subcondition if only the `NoS3StoreAvailable` profile is specified.